### PR TITLE
Add Linear layer in Equinox with all tests passing!

### PR DESCRIPTION
**Pull request by ameya98**

### CODE REVIEW

**Comment by ameya98** [comment]
Not sure why lint is failing :(

File: e3nn_jax/_src/linear_equinox.py [hunk 1]
```diff
@@ -0,0 +1,347 @@
...
+    _linear: FunctionalLinear
+    _weights: Dict[str, jnp.ndarray]
+    _output_irreps_unsimplified: e3nn.Irreps
+    _output_irreps: e3nn.Irreps
+    _input_dtype: jnp.dtype
+
+    def __init__(
+        self,
+        irreps_out: e3nn.Irreps,
+        irreps_in: e3nn.Irreps,
```
Comment (mariogeiger) [inline e3nn_jax/_src/linear_equinox.py:128]: What about in before out?

Comment (ameya98) [reply e3nn_jax/_src/linear_equinox.py:128]: The Flax and Haiku ones have out before in, so I thought it would be better to be consistent

Comment (mariogeiger) [reply e3nn_jax/_src/linear_equinox.py:126]: Flax and Haiku don't require `irreps_in` to be specified. Actually we could even remove `irreps_in` in the Flax and Haiku

File: e3nn_jax/_src/s2grid.py [hunk 2]
```diff
@@ -1229,6 +1229,8 @@ def m0_values_to_irrepsarray(m0_values, lmax, p_val, p_arg) -> e3nn.IrrepsArray:
         lmax + 1
     )
     irreps = s2_irreps(lmax, p_val, p_arg)
+    m0 = jnp.zeros((*m0_values.shape[:-1], (lmax + 1) ** 2))
+    m0 = m0.at[:, m0_indices].set(m0_values)
```
[the remark this answers is not in the document]
Comment (ameya98) [reply e3nn_jax/_src/s2grid.py:1233]: Oops sorry, this was for formatting

File: e3nn_jax/_src/linear_equinox.py (hunk 1 above)
Comment (ameya98) [reply e3nn_jax/_src/linear_equinox.py:126]: But also the convention would change `e3nn.equinox.Linear("1o")` would mean something different from `e3nn.flax.Linear("1o")`. I would prefer to keep it as is.

Comment (mariogeiger) [reply e3nn_jax/_src/linear_equinox.py:126]: `e3nn.equinox.Linear("1o")` would crash right?

But as you wish, I can merge like that

**Pull request merged**